keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/ImageService.kt

```diff
@@ -84,109 +77,42 @@ class ImageService(
       }
 
   /**
-   * Get a specific image for an app version.
-   *
-   * @param region if supplied the latest image in this region is returned, if `null` the latest
-   * image regardless of region.
+   * Returns the latest image that is present in all regions.
+   * Each AMI must have tags.
    */
-  suspend fun getLatestNamedImage(appVersion: AppVersion, account: String, region: String? = null): NamedImage? =
-    cloudDriverService.namedImages(
+  suspend fun getLatestNamedImageWithAllRegionsForAppVersion(appVersion: AppVersion, account: String, regions: Collection<String>): Collection<NamedImage> {
```

Comment:
I'd make this plural `getLatestNamedImages...` since it now returns a collection, and update the header comment to reflect the fact that there are scenarios where it can return multiple NamedImage objects.